### DISCUSSION TITLE
Static IP on a DHCP type IPAM.

### DISCUSSION THREAD

**Discussion by mskarbek** [containers/podman#9438] [Q&A] [answered]
Is there a way to add a container with static IP assigned to the network with bridge IPAM set as DHCP?
CNI config for the network:
```
{
  "cniVersion": "0.4.0",
  "name": "br1",
  "plugins": [
    {
      "type": "bridge",
      "bridge": "br1",
      "ipam": {
        "type": "dhcp"
      }
    }
  ]
}
```
Podman command: `podman run -it --rm --ip 192.168.42.253 --network br1 registry.access.redhat.com/ubi8/ubi:8.3`
Error:
```
ERRO[0152] Error adding network: error calling DHCP.Allocate: no more tries 
ERRO[0152] Error while adding pod to CNI network "br1": error calling DHCP.Allocate: no more tries 
Error: error configuring network namespace for container ff594f652e28cc218332de514a88961cdf55d6d024e5174d09599eaeb231e125: error calling DHCP.Allocate: no more tries
```
I know that this can be done by manually setting up network namespace, but it would be nice if this just worked.
Yes, error clearly tells that there is no DHCP server, but the thing is that podman should completely skip DHCP step if static IP is set. I know that this is more of a CNI plugin issue then podman, but there should be a way to tell CNI plugin to skip DHCP step and that, I think, requires both parties to coordinate.

**Comment by baude** [comment]
@mccv1r0 what do you think?

**Comment by mccv1r0** [comment, marked as answer]
Yes.  To start you really should make sure that the static ip isn't something that the dhcp server is going to hand out.

For podman, you would need 2 .conflist files, each with a different name obviously.  One has ipam `type: dhcp` (like in your example) and the other has ipam `type: host-local`.  They both point to the same Linux bridge.

**Reply by mskarbek** [reply]
That worked perfectly, thanks.
